RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+These peers will forward the joining request, with the peer address, to the rest of the network until CNs with similar node ids have been reached.
+CNs with similar node ids will then add the new peer address of the new node to their routing table, allowing for fast discovery of the new CC or CN. 
+
+Other CCs and CNs will then be able to retrieve the new CCs or CNs peer address by submitting discovery requests.
+Once the peer address of the desired CC or CN has been discovered then a direct P2P communication channel can be established between the two parties for any future communication.
+After discovery, the rest of the Tari communication network will be unaware of any further communication between the two parties.
+
+#### Sending Data Messages and Discovery Requests
+
+The majority of all communication on the Tari communication network will be performed using direct P2P channels established between different CCs and CNs once they are aware of the peer addresses of each other that contain their online communication addresses.
+Message propagation on the network will typically consist only of joining and discovery requests where a CC or CN wants to join the network or retrieve the peer address of another CC or CN so that a direct P2P channel can be established.
+
+Messages can be transmitted in this network in either an unencrypted or encrypted form.
+Typically messages that have been sent in unencrypted form is of interest to a number of CNs on the network and should be propagated so that every CN that is interested in that data message obtains a copy.
+Block and Transaction propagation are examples of data messages where multiple entities on the Tari communication network are interested in that data message, this requires propagation through the entire Tari communication network in unencrypted form.
+
+Encrypted data messages make use of the source and destinations identification cryptographic keys to construct a shared secret with which the message can be encoded and decoded.
+This ensures that only the two parties are able to decode the data message as it is propagated through the communication network.
+This mechanism can be used to perform private discovery requests, where the online communication address of the source node is encrypted and propagated through the network until it reached the destination node.
+Private discovery requests can only be performed if both parties are online at the same time.
+Encryption of the data message ensuring that only the destination node is able to view the online address of the source node as the data message moves through the network.
+Once the destination node receives and decrypts the data message, that node is then able to establish a P2P communication channel with the source node for any further communication.
+
+Propagation of completely private discovery request, hidden as an encrypted data message, can be performed as a broadcast through the entire network using the Gossip protocol.
```

Comment:
One way to limit misuse of the messaging system is to have data messages that have a visible and encrypted part, and a signature.  The source node must have its identification public key visible and sign the entire message (visible and encrypted part). No one on the network will know who the destination node is, but they are able to ban/blacklist the specific source node if needed. The signing of the message will limit users from mimicking other users. Maybe a mechanism like this could work?

Reply:
Sounds good for now